tlsf.c

```diff
@@ -977,45 +977,68 @@ size_t tlsf_alloc_overhead(void)
 	return block_header_overhead;
 }
 
-pool_t tlsf_add_pool(tlsf_t tlsf, void* mem, size_t bytes)
+pool_t tlsf_extend_pool(tlsf_t tlsf, void* mem, size_t bytes, size_t incr)

```

Comment:
API is ambiguous. I'd prefer:
pool_t tlsf_extend_pool(tlsf_t tlsf, pool_t pool, size_t incr);

Reply:
Yes, I also think your prototype is better, but since the alignment may happen, it's impossible to recover the original size by finding the sentinal block.

Reply:
What about:
pool_t tlsf_resize_pool(tlsf_t tlsf, pool_t pool, size_t oldBytes, size_t newBytes)

Reply:
Yes, it's also my initial thinking, but it's very hard(if not possible) to reduce the running pool size since some memory may be using by user. That's why I:

1. Name the new function tlsf_extend_pool
2. Pass the increased bytes instead new size(newBytes may smaller than oldBytes)

Reply:
> it's very hard(if not possible) to reduce the running pool size since some memory may be using by user.

Documentation will *warn* about dangerousness of reducing running pool size.
User anyway should know what he is doing.

Reply:
Ok, looks reasonable.